chain/chunks/src/client.rs

```diff
@@ -105,12 +103,12 @@ impl ShardedTransactionPool {
     pub fn reintroduce_transactions(
         &mut self,
         shard_uid: ShardUId,
-        signed_txs: impl IntoIterator<Item = SignedTransaction>,
+        validated_txs: impl IntoIterator<Item = ValidatedTransaction>,
```

Comment:
side comment - I dislike the whole idea of getting transaction from the pool, then reintroducing them back, then removing them once they're in a block, then potentially introducing them back in in case of a fork. I also dislike doing that in the client. While technically correct I feel like the benefits we could get from simplifying it would greatly outweigh the potential of losing a few transactions in those rare cases. Especially given the existing redundancy in transaction distribution.

Reply:
oh absolutely, Anton A and I have also discussed how this whole thing is unnecessarily complicated and hopefully we can address it at some point.